front/app/modules/commercial/insights/admin/containers/Insights/Details/Network/index.tsx

```diff
@@ -0,0 +1,224 @@
+import { withRouter, WithRouterProps } from 'react-router';
+import React, {
+  useEffect,
+  useRef,
+  useState,
+  useMemo,
+  useCallback,
+} from 'react';
+
+// graph
+import * as d3 from 'd3';
+import ForceGraph2D, {
+  ForceGraphMethods,
+  NodeObject,
+} from 'react-force-graph-2d';
+
+// hooks
+import useNetwork from 'modules/commercial/insights/hooks/useInsightsNetwork';
+
+// types
+import { IInsightsNetworkNode } from 'modules/commercial/insights/services/insightsNetwork';
+
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+import { cloneDeep } from 'lodash-es';
+import { colors } from 'utils/styleUtils';
+
+// components
+import { Box } from 'cl2-component-library';
+import Button from 'components/UI/Button';
+
+type CanvasCustomRenderMode = 'replace' | 'before' | 'after';
+type Node = NodeObject & IInsightsNetworkNode;
+
+const zoomStep = 0.2;
+
+const Network = ({ params: { viewId } }: WithRouterProps) => {
+  const [initialCenter, setInitialCenter] = useState(true);
+  const [height, setHeight] = useState(0);
+  const [width, setWidth] = useState(0);
+  const [zoomLevel, setZoomLevel] = useState(0);
+
+  const [collapsedClusters, setCollapsedClusters] = useState<string[]>([]);
+  const forceRef = useRef<ForceGraphMethods>();
+  const network = useNetwork(viewId);
+
+  useEffect(() => {
+    if (forceRef.current) {
+      forceRef.current.d3Force('charge')?.strength(-25);
+      forceRef.current.d3Force('link')?.distance(40);
+      forceRef.current.d3Force('charge')?.distanceMax(80);
+      forceRef.current.d3Force(
+        'collide',
+        // @ts-ignore
+        d3.forceCollide().radius((node: IInsightsNetworkNode) => {
+          const isClusterNode = node.cluster_id === null;
+          return isClusterNode ? node.val / 4 : node.val * 3 + 8;
+        })
+      );
+    }
+  });
+
+  const clusterIds = useMemo(() => {
+    if (!isNilOrError(network)) {
+      return network.attributes.nodes
+        .filter((node) => node.cluster_id === null)
+        .map((node) => node.id);
+    } else return [];
+  }, [network]);
+
+  useEffect(() => {
+    setCollapsedClusters(clusterIds);
+    setInitialCenter(true);
+  }, [clusterIds]);
+
+  const networkAttributes = useMemo(() => {
+    if (!isNilOrError(network)) {
+      return cloneDeep(network.attributes);
```

Comment:
For some reason, the viz library needs a copy of the object that it can mutate freely in order to render the canvas. Not super clear on the internals of this library but passing the data directly from the hook breaks the functionality and crashes the vizualization.